test/e2e/serial/tests/configuration.go

```diff
@@ -685,9 +689,78 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
 			}
 			Expect(err).ToNot(HaveOccurred(), "failed to check the NodeGroupConfig status for %q", nroKey.String())
 		})
+
+		It("should report relatedObjects in the status", func(ctx context.Context) {
+			nroKey := objects.NROObjectKey()
+			nroOperObj := nropv1.NUMAResourcesOperator{}
+
+			err := fxt.Client.Get(context.TODO(), nroKey, &nroOperObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroKey.String())
+
+			if len(nroOperObj.Spec.NodeGroups) != 1 {
+				// TODO: this is the simplest case, there is no hard requirement really
+				// but we took the simplest option atm
+				e2efixture.Skipf(fxt, "more than one NodeGroup not yet supported, found %d", len(nroOperObj.Spec.NodeGroups))
+			}
+
+			dss, err := objects.GetDaemonSetsOwnedBy(fxt.Client, nroOperObj.ObjectMeta)
+			Expect(err).ToNot(HaveOccurred())
+
+			dssExpected := namespacedNameListToStringList(nroOperObj.Status.DaemonSets)
+			dssGot := namespacedNameListToStringList(daemonSetListToNamespacedNameList(dss))
+			Expect(dssGot).To(Equal(dssExpected), "mismatching RTE DaemonSets for NUMAResourcesOperator")
+
+			// shortcut, they all must be here anyway
+			nroExpected := objRefListToStringList(relatedobjects.ResourceTopologyExporter(dss[0].Namespace, nroOperObj.Status.DaemonSets))
+			nroGot := objRefListToStringList(nroOperObj.Status.RelatedObjects)
+			Expect(nroGot).To(Equal(nroExpected), "mismatching related objects for NUMAResourcesOperator")
+
+			nroSchedKey := objects.NROSchedObjectKey()
+			nroSchedObj := nropv1.NUMAResourcesScheduler{}
+
+			err = fxt.Client.Get(context.TODO(), nroSchedKey, &nroSchedObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroSchedKey.String())
+
+			dps, err := objects.GetDeploymentOwnedBy(fxt.Client, nroSchedObj.ObjectMeta)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(dps)).To(Equal(1), "unexpected amount of scheduler deployments: %d", len(dps))
+
+			nrsExpected := objRefListToStringList(relatedobjects.Scheduler(dps[0].Namespace, nroSchedObj.Status.Deployment))
+			nrsGot := objRefListToStringList(nroSchedObj.Status.RelatedObjects)
+			Expect(nrsGot).To(Equal(nrsExpected), "mismatching related objects for NUMAResourcesScheduler")
```

Comment:
ginkgo reporting can be handy for debugging and understanding what is being done. By() for the 2 main checks (one for NRO and one for the sched) is enough.

Reply:
right! let me add it